app/screens/edit_server/form.test.tsx

```diff
@@ -0,0 +1,261 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {fireEvent, screen} from '@testing-library/react-native';
+import React from 'react';
+
+import {getDefaultThemeByAppearance} from '@context/theme';
+import {renderWithIntlAndTheme} from '@test/intl-test-helper';
+
+import EditServerForm from './form';
+
+const defaultProps = {
+    buttonDisabled: false,
+    connecting: false,
+    displayName: 'Test Server',
+    displayNameError: undefined,
+    handleUpdate: jest.fn(),
+    handleDisplayNameTextChanged: jest.fn(),
+    handlePreauthSecretTextChanged: jest.fn(),
+    handlePreauthSecretFocus: jest.fn(),
+    keyboardAwareRef: {current: null},
+    serverUrl: 'https://example.com',
+    preauthSecret: '',
+    theme: getDefaultThemeByAppearance(),
+};
+
+describe('EditServerForm', () => {
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('should render form with display name input', () => {
+        renderWithIntlAndTheme(<EditServerForm {...defaultProps}/>);
+
+        expect(screen.getByTestId('edit_server_form.server_display_name.input')).toBeTruthy();
+        expect(screen.getByTestId('edit_server_form.display_help')).toBeTruthy();
+        expect(screen.getByTestId('edit_server_form.save.button')).toBeTruthy();
+    });
+
+    it('should render advanced options toggle', () => {
+        renderWithIntlAndTheme(<EditServerForm {...defaultProps}/>);
+
+        expect(screen.getByTestId('edit_server_form.advanced_options.toggle')).toBeTruthy();
+        expect(screen.getByText('Advanced Options')).toBeTruthy();
+    });
+
+    it('should show chevron-down initially', () => {
+        renderWithIntlAndTheme(<EditServerForm {...defaultProps}/>);
+
+        const toggle = screen.getByTestId('edit_server_form.advanced_options.toggle');
+        expect(toggle).toBeTruthy();
+
+        // The chevron icon name is internal to CompassIcon, so we'll just verify the toggle exists
+    });
```

Comment:
This test is not adding anything new to the previous one. If I am not mistaken, the Compass Icon is mocked, so you could find it by test id, and check the prop (I think you can directly call `toHaveProp`). 0/5 whether it makes sense (or whether makes more sense to do this alongside the show/hide test).